paasta_tools/kubernetes_tools.py

```
@@ -0,0 +1,581 @@
+        service=service,
+        instance=instance,
+        cluster=cluster,
+        load_deployments=load_deployments,
+        soa_dir=soa_dir,
+    )
+
+
+class InvalidKubernetesConfig(Exception):
+    pass
+
+
+class KubernetesDeploymentConfig(LongRunningServiceConfig):
+    config_dict: KubernetesDeploymentConfigDict
+
+    config_filename_prefix = 'kubernetes'
+
+    def __init__(
+        self,
+        service: str,
+        cluster: str,
+        instance: str,
+        config_dict: KubernetesDeploymentConfigDict,
+        branch_dict: Optional[BranchDictV2],
+        soa_dir: str=DEFAULT_SOA_DIR,
+    ) -> None:
+        super(KubernetesDeploymentConfig, self).__init__(
+            cluster=cluster,
+            instance=instance,
+            service=service,
+            config_dict=config_dict,
+            branch_dict=branch_dict,
+            soa_dir=soa_dir,
+        )
+
+    def __repr__(self) -> str:
+        return "KubernetesDeploymentConfig({!r}, {!r}, {!r}, {!r}, {!r}, {!r})".format(
+            self.service,
+            self.cluster,
+            self.instance,
+            self.config_dict,
+            self.branch_dict,
+            self.soa_dir,
+        )
+
+    def copy(self) -> "KubernetesDeploymentConfig":
+        return self.__class__(
+            service=self.service,
+            instance=self.instance,
+            cluster=self.cluster,
+            config_dict=dict(self.config_dict),
+            branch_dict=dict(self.branch_dict) if self.branch_dict is not None else None,
+            soa_dir=self.soa_dir,
+        )
+
+    def get_bounce_method(self) -> str:
+        """Get the bounce method specified in the service's kubernetes configuration."""
+        # map existing bounce methods to k8s equivalents.
+        kube_deploy_stategy_map = {'crossover': 'RollingUpdate', 'downthenup': 'Recreate'}
+        return kube_deploy_stategy_map[self.config_dict.get('bounce_method', 'crossover')]
```

worth making this ``kube_deploy_strategy_map.get(...)`` instead, incase a bad value creeps into the config dict?

---

We talked a bit about this IRL. I might try and do some fancy exception handling but I might also just ship this as is and let the KeyError bubble up if it does happen.